subprojects/language-native/src/main/java/org/gradle/language/swift/internal/DefaultSwiftBinary.java

```diff
@@ -150,4 +174,73 @@ public FileCollection getObjects() {
     public Property<SwiftCompile> getCompileTask() {
         return compileTaskProperty;
     }
+
+    @Inject
+    protected NativeDependencyCache getNativeDependencyCache() {
+        throw new UnsupportedOperationException();
+    }
+
+    private class ModulePath implements MinimalFileSet, Buildable {
+        private final Configuration importPathConfig;
+
+        private Set<File> result;
+
+        ModulePath(Configuration importPathConfig) {
+            this.importPathConfig = importPathConfig;
+        }
+
+        @Override
+        public String getDisplayName() {
+            return "Module include path for " + DefaultSwiftBinary.this.toString();
+        }
+
+        @Override
+        public Set<File> getFiles() {
+            if (result == null) {
+                result = Sets.newLinkedHashSet();
+                Map<String, ModuleMap> moduleMaps = Maps.newHashMap();
+                for (ResolvedArtifactResult artifact : importPathConfig.getIncoming().getArtifacts()) {
+                    Usage usage = artifact.getVariant().getAttributes().getAttribute(Usage.USAGE_ATTRIBUTE);
+                    if (usage != null && Usage.C_PLUS_PLUS_API.equals(usage.getName())) {
+                        String moduleName;
+
+                        ComponentIdentifier id = artifact.getId().getComponentIdentifier();
+                        if (ModuleComponentIdentifier.class.isAssignableFrom(id.getClass())) {
+                            moduleName = ((ModuleComponentIdentifier) id).getModule();
+                        } else if (ProjectComponentIdentifier.class.isAssignableFrom(id.getClass())) {
+                            moduleName = ((ProjectComponentIdentifier) id).getProjectName();
+                        } else if (LibraryBinaryIdentifier.class.isAssignableFrom(id.getClass())) {
+                            moduleName = ((LibraryBinaryIdentifier) id).getLibraryName();
+                        } else {
+                            throw new IllegalArgumentException("Could not determine the name of " + id.getDisplayName() + ": unknown component identifier type: " + id.getClass().getSimpleName());
+                        }
+
+                        ModuleMap moduleMap;
+                        if (moduleMaps.containsKey(moduleName)) {
```

Comment:
Doesn't this mean that if I have...

`com.example.foo:MyLib:1.0` and `com.example.bar:MyLib:1.0`, the module names are the same, so we would try to create a single module from them?

Reply:
Yes it does.  The map is intended to deal with multiple public header dir artifacts being published, but with the default of using the component name as the module name in the module map, we'll always have this conflict at some level, I think.  To avoid this, I think we need some better (but still reasonable) way to derive a default module name or some configurability for the module map so the consumer can sort it out.

In any event, I've changed it such that different module maps will be produced in this scenario, but they'll still have the same module name.  Not sure what this will do during compilation.